### PR TITLE
fix(OptimizelyConfig): add more tests for experiment audiences serialization

### PR DESCRIPTION
**Pull request by jaeopt**
- add more tests for audience serialization (empty audiences, NOT conditions)

### CODE REVIEW

File: Tests/OptimizelyTests-DataModel/ExperimentTests.swift [hunk 1]
```diff
@@ -192,6 +196,8 @@ extension ExperimentTests {
             "\"us\" OR \"female\"",
             "(\"us\" OR \"female\") AND \"adult\"",
             "(\"us\" OR (\"female\" AND \"adult\")) AND (\"fr\" AND (\"male\" OR \"kid\"))",
+            "NOT (\"us\" AND \"female\")",
```
Comment (keppel2) [inline Tests/OptimizelyTests-DataModel/ExperimentTests.swift:199]: Use `#"raw "string" which is in quotes"#`?

Comment (jaeopt) [reply Tests/OptimizelyTests-DataModel/ExperimentTests.swift:199]: I didn't know about this trick :) I'll skip it in this case.

Comment (keppel2) [reply Tests/OptimizelyTests-DataModel/ExperimentTests.swift:199]: Swift 5+, no big deal!

**Review by The-inside-man — APPROVED** [review]
lgtm
